tests/test_types.py

```diff
@@ -7,6 +8,48 @@
 from pydargs import parse
 
 
+class TestList:
+    @dataclass
+    class Config:
+        a: list[int]
+        b: list[str] = field(default_factory=lambda: [])
+        c: Sequence[float] = field(default_factory=lambda: [1.0])
+        d: Sequence[str] = ("a", "b")
+        e: str = "dummy"
+
+    def test_a_required(self, capsys):
+        with raises(SystemExit):
+            parse(self.Config, [])
+        captured = capsys.readouterr()
+        assert "the following arguments are required: --a" in captured.err
+
+    def test_default(self):
+        config = parse(self.Config, ["--a", "1"])
+        assert config.a == [1]
+        assert config.b == []
+        assert config.c == [1.0]
+        assert config.d == ("a", "b")
+        assert config.e == "dummy"
+
+    def test_list_args(self):
+        config = parse(self.Config, ["--a", "1", "--b", "a", "b"])
+        assert config.b == ["a", "b"]
```

Comment:
can you add a test case where there is another option after the list? 
something like
```suggestion
    def test_list_args(self):
        config = parse(self.Config, ["--a", "1", "--b", "a", "b", "--e", "something"])
```